hack/build-image/Dockerfile

```diff
@@ -14,30 +14,33 @@
 
 FROM golang:1.12
 
-RUN mkdir -p /go/src/k8s.io && \
-    cd /go/src/k8s.io && \
-    git config --global advice.detachedHead false && \
-    git clone -b kubernetes-1.15.3 https://github.com/kubernetes/code-generator && \
-    git clone -b kubernetes-1.15.3 https://github.com/kubernetes/apimachinery && \
-    # vendor code-generator go modules to be compatible with pre-1.15
-    cd /go/src/k8s.io/code-generator && GO111MODULE=on go mod vendor && \
-    go get -d sigs.k8s.io/controller-tools/cmd/controller-gen && \
-    cd /go/src/sigs.k8s.io/controller-tools && GO111MODULE=on go mod vendor && \
-    go get golang.org/x/tools/cmd/goimports && \
-    cd /go/src/golang.org/x/tools && \
-    git checkout 40a48ad93fbe707101afb2099b738471f70594ec && \
-    go install ./cmd/goimports && \
-    echo chmod -R a+w /go && \
-    # download & install protoc
-    apt-get update && apt-get install -y unzip && \
-    wget --quiet https://github.com/protocolbuffers/protobuf/releases/download/v3.9.1/protoc-3.9.1-linux-x86_64.zip && \
+RUN mkdir -p /go/src/k8s.io
+WORKDIR /go/src/k8s.io
+RUN git config --global advice.detachedHead false
+RUN git clone -b kubernetes-1.15.3 https://github.com/kubernetes/code-generator
+RUN git clone -b kubernetes-1.15.3 https://github.com/kubernetes/apimachinery 
+# vendor code-generator go modules to be compatible with pre-1.15
+WORKDIR  /go/src/k8s.io/code-generator
+# Don't use ENV here because we don't want to disable modules for subsequent commands
+RUN GO111MODULE=on go mod vendor
+RUN go get -d sigs.k8s.io/controller-tools/cmd/controller-gen
+WORKDIR /go/src/sigs.k8s.io/controller-tools
+RUN GO111MODULE=on go mod vendor
+RUN go get golang.org/x/tools/cmd/goimports
+WORKDIR /go/src/golang.org/x/tools
+RUN git checkout 40a48ad93fbe707101afb2099b738471f70594ec
+RUN go install ./cmd/goimports
+RUN echo chmod -R a+w /go
+# NEED TO CHANGE WORKDIR HERE
+WORKDIR /root
+RUN apt-get update
+RUN apt-get install -y unzip
```

Comment:
I think per https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run we want to combine the `apt-get update` and `apt-get install` steps